app/validation/validator.py

```diff
@@ -708,21 +740,69 @@ def _validate_routing_rules_default(rules, block_or_group):
 
         return errors
 
-    def _validate_routing_rule(self, rule, answer_ids_with_group_id, block_or_group):
+    def _validate_routing_rule(
+        self,
+        rule,
+        answer_ids_with_group_id,
+        block_or_group,
+        option_value_to_answer_id_map,
+    ):
         errors = []
 
         rule = rule.get("goto")
         if "when" in rule:
             errors.extend(
                 self._validate_when_rule(
-                    rule["when"], answer_ids_with_group_id, block_or_group["id"]
+                    rule["when"],
+                    answer_ids_with_group_id,
+                    option_value_to_answer_id_map,
+                    block_or_group["id"],
+                )
+            )
+
+        return errors
+
+    @staticmethod
+    def validate_answer_value_in_when_rule(when_rule, option_value_to_answer_id_map):
+        errors = []
+        when_values = when_rule.get("values", [])
+        when_value = when_rule.get("value")
+        if when_value:
+            when_values.append(when_value)
+
+        if not Validator._is_when_rule_answer_value_valid(
+            answer_id=when_rule["id"],
+            when_values=when_values,
+            option_value_to_answer_id_map=option_value_to_answer_id_map,
+        ):
```

Comment:
- This should check each value and generate an error for each one that is missing. 
- This could be simplified to:

```python
option_values = option_value_to_answer_id_map.get(when_rule["id"])
for value in when_values:
   if value not in option_values:
       errors.append(
           Validator._error_message(
               f"Answer value in when rule with answer id `{when_rule['id']}` has an invalid value of `{value}`"
           )
       )
```